### PR TITLE
Pgp reencrypt

### PR DESCRIPTION
**Pull request by janboll**
Adding a new integration `account-notifier`: This should pick up passwords from vault and re-encrypt them with the user provided PGP Key. It will then send it via e-mail to the user, as before done by terraform-users. 

By decoupling this from terraform we can leverage the latest PGP versions available and lift any restrictions opposed by the terraform implementation.

Related ticket: https://issues.redhat.com/browse/APPSRE-4706

### CODE REVIEW

File: pkg/reconcile.go [hunk 1]
```diff
@@ -37,17 +37,94 @@ func newRunnerConfig() *runnerConfig {
...
-	Diff() ([]Diff, error)
-	Reconcile() error
-	Setup() error
+	CurrentState(context.Context, *ResourceInventory) error
+	DesiredState(context.Context, *ResourceInventory) error
+	Reconcile(context.Context, *ResourceInventory) error
+	Setup(context.Context) error
+}
+
+type Action string
```
Comment (suzana-nesic) [inline pkg/reconcile.go:46]: Is this line needed? Tried looking through to see if it's being referenced anywhere but I can't seem to find it

Comment (janboll) [reply pkg/reconcile.go]: Good spot! deleted

File: internal/account_notifier.go [hunk 2]
```diff
@@ -0,0 +1,387 @@
...
+	}
+
+	return nil
+}
+
+func (n *AccountNotifier) newNotifier(receipt string) *notify.Notify {
+	notifier := notify.New()
+	email := mail.New(n.smtpauth.sender, fmt.Sprintf("%s:%s", n.smtpauth.server, n.smtpauth.port))
+	// Todo: replace jboll@redhat.com with receipt
+	email.AddReceivers("jboll@redhat.com")
```
[the remark this answers is not in the document]
Comment (janboll) [reply internal/account_notifier.go]: fixed!

**Review by suzana-nesic — APPROVED** [review]

**Comment by suzana-nesic** [comment]
LGTM!

**Pull request merged**